test/org/opensolaris/opengrok/configuration/RuntimeEnvironmentTest.java

```diff
@@ -943,48 +944,42 @@ public void testCanAcceptMessage() throws Exception {
     }
 
     @Test
-    public void testLoadEmptyStatistics() throws IOException, ParseException {
+    public void testLoadEmptyStatistics() throws IOException {
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         String json = "{}";
         try (InputStream in = new StringInputStream(json)) {
             env.loadStatistics(in);
         }
-        Assert.assertEquals(new Statistics().toJson(), env.getStatistics().toJson());
+        Assert.assertEquals(new Statistics().encode(), env.getStatistics().encode());
     }
 
     @Test
-    public void testLoadStatistics() throws IOException, ParseException {
+    public void testLoadStatistics() throws IOException {
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         String json = "{"
-            + "\"requests_per_minute_max\":3,"
+            + "\"requestsPerMinuteMax\":3,"
             + "\"timing\":{"
                 + "\"*\":2288,"
                 + "\"xref\":53,"
                 + "\"root\":2235"
             + "},"
             + "\"minutes\":756,"
-            + "\"timing_min\":{"
+            + "\"timingMin\":{"
```

Comment:
TIMING_MIN does not indicate this, am I missing something?

Reply:
i guess there is a magic in the gson stuff?

Reply:
This is encoded Statistics which reflect the attribute names in the class.

Reply:
This is encoded Statistics which reflect the attribute names in the class.

Reply:
No magic, reflection :D

Reply:
```java
            obj.add(REQUEST_CATEGORIES, context.serialize(stats.getRequestCategories()));
+            obj.add(TIMING, context.serialize(stats.getTiming()));
+            obj.add(TIMING_MIN, context.serialize(stats.getTimingMin()));
+            obj.add(TIMING_MAX, context.serialize(stats.getTimingMax()));
+            obj.add(TIMING_AVG, context.serialize(stats.getTimingAvg()));
```

I thought that these are the json fields. But this goes only for the output or what?

Reply:
Yes, there is difference between saving the statistics (all the raw data) vs reporting (processing raw data to have meaningful output -> e.g. averages).

Reply:
aha ok,

but what if user wants to store current statistics in a file and reload them afterwards to reset to grok state after let's say some testing or something like that?

That's not possible now, is it?

Reply:
And I'm sorry I didn't read the PR message :-D

Reply:
Well, all the save/load methods are still in the `RuntimeEnvironment`. However, `StatsMessage` so far provides only `clean` or `reload` but not `save` commands.

Reply:
My usecase was something like
```bash
$ Messages -n stats get > stats.json
$ # doing some requests ...
$ mv stats.json /var/opengrok/etc/statistics.json
$ Messages -n stats reload
```

But it's not a common use case for statistics so maybe this is the way to go. @vladak ?

Reply:
Well, this is probably for testing and requires somehow knowledge of the inner workings. Therefore, you can just turn off/restart the webapp, copy `/var/opengrok/etc/statistics.json`, then do requests and copy it back.

Reply:
But we can add `save` command to `StatsMessage`...